dockers/tpu-extras/Dockerfile

```diff
@@ -0,0 +1,65 @@
+FROM google/cloud-sdk:slim
+
+# This Dockerfile installs pytorch/xla 3.7 wheels. There are also 3.6 wheels available; see below.
+ARG PYTHON_VERSION=3.7
+
+SHELL ["/bin/bash", "-c"]
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        python${PYTHON_VERSION} \
+        python${PYTHON_VERSION}-dev \
+        $( [ ${PYTHON_VERSION%%.*} -ge 3 ] && echo "python${PYTHON_VERSION%%.*}-distutils" ) \
+        build-essential \
+        cmake \
+        wget \
+        unzip \
+        ca-certificates \
+        libomp5 \
+    && \
+
+# Install python dependencies
+    wget https://bootstrap.pypa.io/get-pip.py --progress=bar:force:noscroll --no-check-certificate && \
+    python${PYTHON_VERSION} get-pip.py && \
+    rm get-pip.py && \
+
+# Set the default python and install PIP packages
+    update-alternatives --install /usr/bin/python${PYTHON_VERSION%%.*} python${PYTHON_VERSION%%.*} /usr/bin/python${PYTHON_VERSION} 1 && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python${PYTHON_VERSION} 1 && \
+
+# Install Pytorch XLA
+    py_version=${PYTHON_VERSION/./} && \
+    # Python 3.7 wheels are available. Replace cp36-cp36m with cp37-cp37m
+    gsutil cp "gs://tpu-pytorch/wheels/torch-nightly-cp${py_version}-cp${py_version}m-linux_x86_64.whl" . && \
+    gsutil cp "gs://tpu-pytorch/wheels/torch_xla-nightly-cp${py_version}-cp${py_version}m-linux_x86_64.whl" . && \
+    gsutil cp "gs://tpu-pytorch/wheels/torchvision-nightly-cp${py_version}-cp${py_version}m-linux_x86_64.whl" . && \
+    pip install "torch-nightly-cp${py_version}-cp${py_version}m-linux_x86_64.whl" && \
+    pip install "torch_xla-nightly-cp${py_version}-cp${py_version}m-linux_x86_64.whl" && \
+    pip install "torchvision-nightly-cp${py_version}-cp${py_version}m-linux_x86_64.whl" && \
+    rm "torch-nightly-cp${py_version}-cp${py_version}m-linux_x86_64.whl" && \
+    rm "torch_xla-nightly-cp${py_version}-cp${py_version}m-linux_x86_64.whl" && \
+    rm "torchvision-nightly-cp${py_version}-cp${py_version}m-linux_x86_64.whl" && \
+    pip install mkl && \
+
+# Cleaning
+    apt-get autoremove -y && \
+    apt-get clean
+
+# Install pytorch-lightning dependencies.
+RUN \
+# Get package
+    wget https://github.com/PyTorchLightning/pytorch-lightning/archive/master.zip --progress=bar:force:noscroll && \
```

Comment:
It looks like this PR is using the archived `master` version of pytorch-lightning code. This might be OK for post-submit CI runs but for any pre-submit CI, your test would not have the latest pytorch-lightning code.

I used this strategy to get the current PR's pending code before running tests:
```
# Install pytorch-lightning at the current PR, plus dependencies.
RUN git clone https://github.com/PyTorchLightning/pytorch-lightning.git && \
    cd pytorch-lightning && \
    git fetch origin $GITHUB_REF:CI && \
    git checkout CI && \
    cd .. && \
    pip install ./pytorch-lightning
```

`$GITHUB_REF` was a Docker `ARG` passed in

Reply:
If your plan is to checkout `$GITHUB_REF` in the TPU Dockerfile, then I think it is fine. The only risk is if the pending PR is changing something in `requirements/`

Reply:
no, this fine as it creates kind of baseline, and each PR docker image will still have the requirements install but as most (most likely all) will be satisfied it won't install anything extra :rabbit: